NSURL.m

```diff
@@ -156,6 +157,19 @@ static void posixError(CFErrorRef *error) {
     CFRelease(err);
 }
 
+static void dlError(CFErrorRef *error){
+    const CFStringRef keys[1] = {
+        kCFErrorUnderlyingErrorKey,
+    };
```

Comment:
Here's one little "trick": when a C function expects an array, but you're only passing a single item, you don't have to actually declare it as an array, simply passing a pointer to a single value would do just as well. So:

```suggestion
    const CFStringRef key = kCFErrorUnderlyingErrorKey;
```

and same below for `values` -> `value`.